homeassistant/components/apsystems_local/coordinator.py

```diff
@@ -0,0 +1,95 @@
+"""The coordinator for APsystems local API integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from time import monotonic
+
+from aiohttp import client_exceptions
+from APsystemsEZ1 import APsystemsEZ1M, ReturnOutputData
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InverterNotAvailable(Exception):
+    """Error used when Device is offline."""
+
+
+class ApSystemsDataCoordinator(DataUpdateCoordinator):
+    """Coordinator used for all sensors."""
+
+    def __init__(self, hass: HomeAssistant, api: APsystemsEZ1M) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name="APSystems Data",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=timedelta(seconds=12),
+        )
+        self.api = api
+        self.always_update = True
+
+    async def _async_update_data(self) -> ReturnOutputData | None:  # type: ignore[override]
+        try:
+            return await self.api.get_output_data()
+        except (TimeoutError, client_exceptions.ClientConnectionError):
```

Comment:
These exceptions are already handled, why create your own?

Reply:
But that spams the logs when the inverter is offline, as that will happen every night. So that's why I throw this custom error, to prevent a log spam.